### ISSUE TITLE
Multitenant upgrade fail - no object with type codeunit id 2 found

### ISSUE THREAD

**Issue by ERPgruppenMW** [MicrosoftDocs/dynamics365smb-devitpro-pb#995]
I am upgrading a NAV 2016 tenant through BC 14 and then BC 15. 
Until step 9 everything ok.
Start-Navdataupgrade initiates ok, but gives error:

SessionId         : 0
CodeunitId        : 2
FunctionName      : OnRun
CompanyName       : &lt;Mycompanyname&gt;
StartTime         : 
Duration          : 
State             : FailedPendingResume
Error             : Der blev ikke fundet et objekt af typen CodeUnit med id'et 2.
(Translation: An object of type CodeUnit with Id 2 not found)
ExecutionSourceId : 00000000-0000-0000-0000-000000000000
PostUpgrade       : False

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 5e435e27-0b74-7dcd-555d-977517fd3774
* Version Independent ID: ea922161-12f9-3f86-69f9-2044369d6574
* Content: [Upgrade an unmodified application - Business Central](https://docs.microsoft.com/en-us/dynamics365/business-central/dev-itpro/upgrade/upgrade-unmodified-application)
* Content Source: [dev-itpro/upgrade/upgrade-unmodified-application.md](https://github.com/MicrosoftDocs/dynamics365smb-devitpro-pb/blob/live/dev-itpro/upgrade/upgrade-unmodified-application.md)
* Service: **dynamics365-business-central**
* GitHub Login: @jswymer
* Microsoft Alias: **jswymer**

**Comment by jswymer** [comment]
Hello,

We have seen this issue. It is usually fixed by restarting the BC server instance after publishing the extensions (step 7). What do you see if you run Get-NAVDataUpgrade BC150 -Tenant- ID -Detailed?

**Comment by ERPgruppenMW** [comment]
Hey,
I have a BC Multitenant App and instance running already, and just mount my BC 14 tenant to it. The server instance has been restartet a number of times before mounting tenant.
Get-NavDataupgrade ... -Detailed output below 
ModuleName        : BASE
WorkflowStep      : CheckPreconditions
ExecutionSourceId : 00000000-0000-0000-0000-000000000000
SessionId         : 55
CompanyName       : <Company name 1>
StartTime         : 12-02-2020 09:04:52
Duration          : 00:00:00
State             : Completed
Error             : 
PostUpgrade       : False

ModuleName        : BASE
WorkflowStep      : CheckPreconditions
ExecutionSourceId : 00000000-0000-0000-0000-000000000000
SessionId         : 55
CompanyName       : <Company name 2>
StartTime         : 12-02-2020 09:04:52
Duration          : 00:00:00
State             : Completed
Error             : 
PostUpgrade       : False

ModuleName        : BASE
WorkflowStep      : CheckPreconditions
ExecutionSourceId : 00000000-0000-0000-0000-000000000000
SessionId         : 55
CompanyName       : 
StartTime         : 12-02-2020 09:04:52
Duration          : 00:00:00
State             : Completed
Error             : 
PostUpgrade       : False

ModuleName        : BASE
WorkflowStep      : OnUpgrade
ExecutionSourceId : 00000000-0000-0000-0000-000000000000
SessionId         : 55
CompanyName       : <Company name 1>
StartTime         : 12-02-2020 09:04:52
Duration          : 00:00:00
State             : Completed
Error             : 
PostUpgrade       : False

ModuleName        : BASE
WorkflowStep      : OnUpgrade
ExecutionSourceId : 00000000-0000-0000-0000-000000000000
SessionId         : 55
CompanyName       : <Company name 2>
StartTime         : 12-02-2020 09:04:52
Duration          : 00:00:00
State             : Completed
Error             : 
PostUpgrade       : False

ModuleName        : BASE
WorkflowStep      : OnUpgrade
ExecutionSourceId : 00000000-0000-0000-0000-000000000000
SessionId         : 55
CompanyName       : 
StartTime         : 12-02-2020 09:04:52
Duration          : 00:00:00
State             : Completed
Error             : 
PostUpgrade       : False

ModuleName        : BASE
WorkflowStep      : ValidateUpgrade
ExecutionSourceId : 00000000-0000-0000-0000-000000000000
SessionId         : 55
CompanyName       : <Company name 1>
StartTime         : 12-02-2020 09:04:52
Duration          : 00:00:00
State             : Completed
Error             : 
PostUpgrade       : False

ModuleName        : BASE
WorkflowStep      : ValidateUpgrade
ExecutionSourceId : 00000000-0000-0000-0000-000000000000
SessionId         : 55
CompanyName       : <Company name 2>
StartTime         : 12-02-2020 09:04:52
Duration          : 00:00:00
State             : Completed
Error             : 
PostUpgrade       : False

ModuleName        : BASE
WorkflowStep      : ValidateUpgrade
ExecutionSourceId : 00000000-0000-0000-0000-000000000000
SessionId         : 55
CompanyName       : 
StartTime         : 12-02-2020 09:04:52
Duration          : 00:00:00
State             : Completed
Error             : 
PostUpgrade       : False

ModuleName        : BASE
WorkflowStep      : AfterUpgradeCommit
ExecutionSourceId : 00000000-0000-0000-0000-000000000000
SessionId         : 0
CompanyName       : <Company name 1>
StartTime         : 
Duration          : 
State             : Pending
Error             : 
PostUpgrade       : False

ModuleName        : BASE
WorkflowStep      : AfterUpgradeCommit
ExecutionSourceId : 00000000-0000-0000-0000-000000000000
SessionId         : 0
CompanyName       : <Company name 2>
StartTime         : 
Duration          : 
State             : Pending
Error             : 
PostUpgrade       : False

ModuleName        : BASE
WorkflowStep      : AfterUpgradeCommit
ExecutionSourceId : 00000000-0000-0000-0000-000000000000
SessionId         : 0
CompanyName       : 
StartTime         : 
Duration          : 
State             : Pending
Error             : 
PostUpgrade       : False

SessionId         : 0
CodeunitId        : 2
FunctionName      : OnRun
CompanyName       : <Company name 1>
StartTime         : 
Duration          : 
State             : FailedPendingResume
Error             : Der blev ikke fundet et objekt af typen CodeUnit med id'et 2.
ExecutionSourceId : 00000000-0000-0000-0000-000000000000
PostUpgrade       : False

SessionId         : 0
CodeunitId        : 2
FunctionName      : OnRun
CompanyName       : <Company name 2>
StartTime         : 
Duration          : 
State             : Pending
Error             : 
ExecutionSourceId : 00000000-0000-0000-0000-000000000000
PostUpgrade       : False

**Comment by jswymer** [comment]
It seems like the base application was not installed. What do you get if you run Get-NAVAppInfo -ServerInstance [Instance] -Tenant [ID]?

**Comment by ERPgruppenMW** [comment]
You are correct, no apps are installed on tenant.
I only perform step 8 when mounting my BC 14 tenant. Here we only sync 
Sync-NAVApp -ServerInstance <server instance name> -Tenant <tenant ID> -Name "System Application" -Version <extension version>
Sync-NAVApp -ServerInstance <server instance name> -Tenant <tenant ID> -Name "Base Application" -Version <extension version>

My Powershell script is like this:
Import-Module 'C:\Program Files\Microsoft Dynamics 365 Business Central\150\Service\NavAdminTool.ps1'
Mount-NAVTenant $Ins -Tenant $ten -AllowAppDatabaseWrite (databasename etc removed)
Sync-NAVTenant $Ins -Tenant $ten -Mode Sync
Import-NAVServerLicense $Ins -Tenant $ten -Database Tenant -LicenseData (licensefilename removed)
Sync-NAVApp -ServerInstance $Ins -Tenant $ten -Name "System Application" -Version '15.0.36560.36649'
Sync-NAVApp -ServerInstance $Ins -Tenant $ten -Name "Base Application" -Version '15.0.36560.36649'
Start-NAVDataUpgrade -ServerInstance $Ins -Tenant $ten -FunctionExecutionMode Serial

**Comment by jswymer** [comment]
Hmm. Just a thought, but perhaps the base application is not configured correctly in the DestinationAppsForMigration setting (task 4): 

Set-NAVServerConfiguration -ServerInstance <server instance name> -KeyName "DestinationAppsForMigration" -KeyValue '[{"appId":"63ca2fa4-4f03-4f2b-a480-172fef340d3f", "name":"System Application", "publisher": "Microsoft"},{"appId":"437dbf0e-84ff-417a-965d-ed2bb9650972", "name":"Base Application", "publisher": "Microsoft"}]'

With this set, the base application should "installed" when you run Start-NAVDataUpgrade.

**Comment by ERPgruppenMW** [comment]
B-I-N-G-O !! That did the trick. :-D
I did not go through step 1-8 in this guide, as I had a Multitenant solution and I just took a BC DVD and startet my environment up from.

So steps to fix was:
Set-NAVServerConfiguration -ServerInstance -KeyName "DestinationAppsForMigration" -KeyValue '[{"appId":"63ca2fa4-4f03-4f2b-a480-172fef340d3f", "name":"System Application", "publisher": "Microsoft"},{"appId":"437dbf0e-84ff-417a-965d-ed2bb9650972", "name":"Base Application", "publisher": "Microsoft"}]'
Restart-Navserverinstance 
Mount, sync, dataupgrade


Thank you very much for great and fast support on this issue.
